app/services/from_address_creation.rb

```diff
@@ -6,6 +6,7 @@ class FromAddressCreation
 
   def save
     return if from_address.invalid?
+    return if FromAddress.find_by(service_id: from_address.service_id).email_address == from_address.email_address
```

Comment:
```suggestion
   def saved_from_address 
     @saved_from_address ||= FromAddress.find_by(
        service_id: from_address.service_id
      ).email_address
    end
```
Should we move the left half to a private method and memoize the call to the DB?